modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java

```diff
@@ -362,11 +362,15 @@ private void replace(String format, StringBuffer result, MessageContext synCtx)
                     if (mediaType.equals(JSON_TYPE) && inferReplacementType(replacementEntry).equals(STRING_TYPE) &&
                             (!trimmedReplacementValue.startsWith("{") && !trimmedReplacementValue.startsWith("["))) {
                         replacementValue = escapeSpecialChars(replacementValue);
-                        // skip double quotes if replacement is boolean or null or valid json number
-                        if (!trimmedReplacementValue.equals("true") && !trimmedReplacementValue.equals("false")
-                            && !trimmedReplacementValue.equals("null") && !validJsonNumber
-                                .matcher(trimmedReplacementValue).matches()) {
-                            replacementValue = "\"" + replacementValue + "\"";
+                        // Check for following property which will force the string to include quotes
+                        Object force_string_quote = synCtx.getProperty("QUOTE_STRING_IN_PAYLOAD_FACTORY_JSON");
+                        if (force_string_quote != null && ((String) force_string_quote).equalsIgnoreCase("true")) {
+                            // skip double quotes if replacement is boolean or null or valid json number
+                            if (!trimmedReplacementValue.equals("true") && !trimmedReplacementValue.equals("false")
```

Comment:
Merging might not be needed because the nested if does not need to be checked if the property is not available/value, is not true